### PR TITLE
Rename some classes according to Retrofit conventions

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: clients/osv/src/main/kotlin/OsvClient.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import java.util.concurrent.atomic.AtomicReference
+
+import okhttp3.OkHttpClient
+
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+/**
+ * This class wraps the OSV Rest API client in order to make its use simpler and less error-prone.
```
Comment (fviernau) [inline clients/osv/src/main/kotlin/OsvClient.kt:34]: This should now say: "..wraps the OSV service"

File: clients/osv/src/main/kotlin/OsvClient.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ */
+class OsvClient(serverUrl: String? = null, httpClient: OkHttpClient? = null) {
+    private val client = OsvService.create(serverUrl, httpClient)
```
Comment (fviernau) [inline clients/osv/src/main/kotlin/OsvClient.kt:37]: this should now be named `service`.

File: clients/osv/src/main/kotlin/OsvService.kt [hunk 3]
```diff
@@ -19,111 +19,136 @@
...
-        } else {
-            Result.failure(IOException(response.message()))
+interface OsvService {
+    companion object {
+        const val BATCH_REQUEST_MAX_SIZE = 1000
+
+        val JSON = Json { namingStrategy = JsonNamingStrategy.SnakeCase }
+
+        /**
+         * Create an OsvApiClient instance for communicating with the given [server], optionally using a pre-built
```
Comment (fviernau) [inline clients/osv/src/main/kotlin/OsvService.kt:53]: Also `OsvApiClient` is obsolete now.

File: plugins/advisors/osv/src/main/kotlin/Osv.kt [hunk 4]
```diff
@@ -69,7 +69,7 @@ class Osv(name: String, config: OsvConfiguration) : AdviceProvider(name) {
 
     override val details: AdvisorDetails = AdvisorDetails(providerName, enumSetOf(AdvisorCapability.VULNERABILITIES))
 
-    private val service = OsvService(
+    private val service = OsvClient(
```
Comment (fviernau) [inline plugins/advisors/osv/src/main/kotlin/Osv.kt:72]: here as well

**Review by fviernau — CHANGES_REQUESTED** [review]

File: clients/osv/src/main/kotlin/OsvClient.kt [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import okhttp3.OkHttpClient
+
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+/**
+ * This class wraps the OSV Rest API client in order to make its use simpler and less error-prone.
+ */
+class OsvClient(serverUrl: String? = null, httpClient: OkHttpClient? = null) {
```
Comment (fviernau) [inline clients/osv/src/main/kotlin/OsvClient.kt:36]: As you're renaming this class, I believe a better name would be `OsvServiceWrapper`. It's inline with the intention and the already existing class docs.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: clients/osv/src/main/kotlin/OsvClient.kt (hunk 5 above)
Comment (sschuberth) [reply clients/osv/src/main/kotlin/OsvClient.kt:36]: IIUC, per definition the code that wraps the service *is* the client, so I'd prefer to keep the shorter (and more commonly used) "client".

File: clients/bazel-module-registry/src/funTest/kotlin/BazelModuleRegistryClientFunTest.kt [hunk 6]
```diff
@@ -25,7 +25,7 @@ import io.kotest.matchers.comparables.shouldBeGreaterThan
 import io.kotest.matchers.shouldBe
 
 class BazelModuleRegistryClientFunTest : WordSpec({
-    val client = BazelModuleRegistryClient.create()
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply clients/bazel-module-registry/src/funTest/kotlin/BazelModuleRegistryClientFunTest.kt:28]: > couldn't find any indication that the `Service` suffix is a convention.

It's used in their official example code as part of the introduction, which is good enough for me to confirm a convention.

Other than that, "Service" is also what's mostly used in [Retrofit's own code base](https://github.com/search?q=repo%3Asquare%2Fretrofit+%22interface+%22+language%3AKotlin&type=code). The only other suffix I see is "Api", which would be fine with me as well, but as most of our existing code already uses "Service", I've aligned on that.

[the remark this answers is not in the document]
Comment (fviernau) [reply clients/bazel-module-registry/src/funTest/kotlin/BazelModuleRegistryClientFunTest.kt:28]: > It's used in their official example code as part of the introduction, which is good enough for me to confirm a convention.

For me a single use of a suffix in example code is not sufficient proof of a convention. However, would you be ok to change the rationale to align on "Service" for consistency within ORT's code base?

File: clients/osv/src/main/kotlin/OsvClient.kt (hunk 5 above)
Comment (fviernau) [reply clients/osv/src/main/kotlin/OsvClient.kt:36]: > IIUC, per definition the code that wraps the service is the client,

I see, so what is the code that _uses_, not _wraps_ the service then?
To me it seems that client would better fit to the code using that wrapper. 

> so I'd prefer to keep the shorter (and more commonly used) "client".

ok, normally I don't care too much, in this case I would appreciate if the compromise was made towards my preference.

File: clients/osv/src/funTest/kotlin/OsvServiceFunTest.kt [hunk 7]
```diff
@@ -66,21 +66,21 @@ class OsvServiceFunTest : StringSpec({
     "getVulnerabilitiesForPackage() returns the expected vulnerability when queried by commit" {
```
Comment (fviernau) [inline clients/osv/src/funTest/kotlin/OsvServiceFunTest.kt:66]: OsvServiceFunTest -> OsvClientFunTest

**Review by fviernau — CHANGES_REQUESTED** [review]

File: clients/bazel-module-registry/src/funTest/kotlin/BazelModuleRegistryClientFunTest.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (sschuberth) [reply clients/bazel-module-registry/src/funTest/kotlin/BazelModuleRegistryClientFunTest.kt:28]: > For me a single use of a suffix in example code is not sufficient proof of a convention.

I've also provided a link to multiple other examples in Retrofit's code base.

> However, would you be ok to change the rationale to align on "Service" for consistency within ORT's code base?

Sure, if that rationale works better for you.

File: clients/osv/src/main/kotlin/OsvClient.kt (hunk 5 above)
Comment (sschuberth) [reply clients/osv/src/main/kotlin/OsvClient.kt]: I've checked the code base again, and OSV seems to currently be the only client library project that providers a higher-level "wrapper" around the low-level `Service` interface. In my thinking that "wrapper" is something between the service and the (advisor) client, so going for "ServiceWrapper" here is fine with me.

Comment (sschuberth) [reply clients/osv/src/main/kotlin/OsvClient.kt]: Or actually, thinking about it again, we could also keep `OsvServiceWrapper` at `OsvClient` to keep thing simple, as you prefer.

File: clients/bazel-module-registry/src/funTest/kotlin/BazelModuleRegistryClientFunTest.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (fviernau) [reply clients/bazel-module-registry/src/funTest/kotlin/BazelModuleRegistryClientFunTest.kt:28]: Thanks!

File: clients/osv/src/main/kotlin/OsvClient.kt (hunk 5 above)
Comment (fviernau) [reply clients/osv/src/main/kotlin/OsvClient.kt]: I'd prefer to keep `OsvServiceWrapper` as it is right now.

**Review by fviernau — APPROVED** [review]

**Pull request merged**